src/compiler/target/golang/mod.rs

```diff
@@ -0,0 +1,268 @@
+
+    let mut code = Self {};
+
+    // define functions
+    for (_, func) in t.functions {
+      code.function(func, lb);
+    }
+
+    // define types
+    for (_, type_) in t.types {
+      code.code_type(type_, lb);
+    }
+
+    // define structs
+    for (_, structure) in t.structs {
+      code.structure(structure, lb);
+    }
+    
+    // define globals
+    for (_, glob) in t.vars {
+      code.global_var(glob, lb);
+    }
+
+    Ok(())
+  }
+  /// Get the type as a string
+  pub fn get_type(&mut self, type_: Option<tokenize::types::Type>) -> String {
+    match type_ {
+      Some(res) => {
+        match res.type_ {
+          tokenize::types::TypeType::Char => "rune".to_string(),
+          tokenize::types::TypeType::String => "string".to_string(),
+          tokenize::types::TypeType::Struct(res) => self.type_struct(res),
+          tokenize::types::TypeType::Int => "int".to_string(),
+          tokenize::types::TypeType::I8 => "int8".to_string(),
+          tokenize::types::TypeType::I16 => "int16".to_string(),
+          tokenize::types::TypeType::I32 => "int32".to_string(),
+          tokenize::types::TypeType::I64 => "int64".to_string(),
+          tokenize::types::TypeType::UInt => "uint".to_string(),
+          tokenize::types::TypeType::U8 => "uint8".to_string(),
+          tokenize::types::TypeType::U16 => "uint16".to_string(),
+          tokenize::types::TypeType::U32 => "uint32".to_string(),
+          tokenize::types::TypeType::U64 => "uint64".to_string(),
+          tokenize::types::TypeType::TypeRef(res) => res,
+          _ => unimplemented!()
+        }
+      },
+      None => "".to_string()
+    }
+  }
+  /// Parse a custom type definition
+  pub fn code_type(&mut self, type_: tokenize::types::GlobalType, lb: &mut impl BuildItems) {
+    lb.code(format!("type {} {}", type_.name, self.get_type(Some(type_.type_))));
+  }
+  /// Parse a function
+  pub fn function(&mut self, func: Function, lb: &mut impl BuildItems) {
+    let mut prefix_str = format!("func {}(", func.name.unwrap());
+    let mut args = vec![];
+    for (name, type_) in func.args {
+      args.push(format!("{} {}", name, self.get_type(Some(type_))));
+    }
+    prefix_str += &args.join(", ");
+    prefix_str += ")";
+
+    prefix_str += " ";
+    prefix_str += &self.get_type(func.res);
+
+    let mut actions = Block::new();
+    for action in func.body.list {
+      self.action(action, &mut actions, false);
+    }
+
+    lb.function(Inline::from_str(prefix_str), actions);
+  }
+  /// Parse a const variable
+  pub fn global_var(&mut self, var: Variable, lb: &mut impl BuildItems) {
+    let mut inline = Inline::new();
+
+    inline.code(format!("const {} = ", var.name.to_string()));
+    self.action(*var.action, &mut inline, true);
+
+    lb.inline(inline);
+  }
+  /// Parse a structure
+  pub fn structure(&mut self, structure: Struct, lb: &mut impl BuildItems) {
+    let prefix_str = format!("type {} struct ", structure.name.unwrap());
```

Comment:
I think this will fail if the user defines a struct within a struct.  
like: 
```go
struct foo {
  bar struct {} // This struct is inline
}
```

I think it's best to output `type {} struct ` if it's not inline and if so only return `struct `